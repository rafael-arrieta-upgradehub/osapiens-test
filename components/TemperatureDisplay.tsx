import React from 'react';
import { View, StyleSheet } from 'react-native';
import { ThemedText } from '@/components/ThemedText';

interface TemperatureDisplayProps {
    temperature: number;
    weatherDescription: string;
}

const TemperatureDisplay: React.FC<TemperatureDisplayProps> = ({ temperature, weatherDescription }) => {
    return (
        <View style={styles.temperatureContainer}>
            <ThemedText style={styles.temperature}>{temperature.toFixed(2)}Â°</ThemedText>
            <ThemedText style={styles.weather}>{weatherDescription}</ThemedText>
        </View>
    );
};

const styles = StyleSheet.create({
    temperatureContainer: {
        alignItems: 'center',
    },
    temperature: {
        fontSize: 100,
        fontWeight: 'bold',
    },
    weather: {
        paddingTop: 36,
        fontSize: 20,
        textTransform: 'capitalize',
    },
});

export default TemperatureDisplay;
