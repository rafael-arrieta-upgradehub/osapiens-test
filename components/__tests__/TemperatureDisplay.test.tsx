import React from 'react';
import renderer from 'react-test-renderer';
import TemperatureDisplay from '@/components/TemperatureDisplay';

describe('TemperatureDisplay', () => {
    it('renders correctly with temperature and weather description', () => {
        const tree = renderer.create(
            <TemperatureDisplay temperature={25} weatherDescription="Warm" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
    it('renders correctly with temperature and weather description', () => {
        const tree = renderer.create(
            <TemperatureDisplay temperature={25} weatherDescription="Cold" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
