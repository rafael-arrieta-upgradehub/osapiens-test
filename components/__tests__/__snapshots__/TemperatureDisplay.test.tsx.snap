// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureDisplay renders correctly with temperature and weather description 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 100,
          "fontWeight": "bold",
        },
      ]
    }
  >
    25.00
    °
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "paddingTop": 36,
          "textTransform": "capitalize",
        },
      ]
    }
  >
    Warm
  </Text>
</View>
`;

exports[`TemperatureDisplay renders correctly with temperature and weather description 2`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 100,
          "fontWeight": "bold",
        },
      ]
    }
  >
    25.00
    °
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "paddingTop": 36,
          "textTransform": "capitalize",
        },
      ]
    }
  >
    Cold
  </Text>
</View>
`;
